File: easy/1732. Find the Highest Altitude/1732. Find the Highest Altitude.py
from typing import List


class Solution:
    def largestAltitude(self, gain: List[int]) -> int:
        altitudes = [0] * (len(gain) + 1)

        for i in range(len(gain)):
            altitudes[i + 1] = altitudes[i] + gain[i]

        return max(altitudes)

File: easy/1732. Find the Highest Altitude/1732. Find the Highest Altitude.go
package main

import "slices"

func largestAltitude(gain []int) int {
	n := len(gain)
	altitudes := slices.Repeat([]int{0}, n+1)

	for i := 0; i < n; i++ {
		altitudes[i+1] = altitudes[i] + gain[i]
	}

	return slices.Max(altitudes)
}
